hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/file/TestOMRecoverLeaseRequest.java

```diff
@@ -297,6 +298,40 @@ public void testRecoverAbsentFile() throws Exception {
     verifyTables(false, false);
   }
 
+  @Test
+  public void testLeaseSoftLimitForHsyncRecoverFile() throws Exception {
+    populateNamespace(true, true, true, true);
+
+    // update soft limit to high value
+    ozoneManager.getConfiguration().set(OzoneConfigKeys.OZONE_OM_LEASE_SOFT_LIMIT, "2s");
+    // Call recovery inside soft limit period it should fail
+    OMClientResponse omClientResponse = validateAndUpdateCache();
+    OMResponse omResponse = omClientResponse.getOMResponse();
+    Assertions.assertEquals(OzoneManagerProtocolProtos.Status.KEY_UNDER_LEASE_SOFT_LIMIT_PERIOD,
+        omResponse.getStatus());
+    // Call second time inside soft limit period also should fail
+    omClientResponse = validateAndUpdateCache();
+    omResponse = omClientResponse.getOMResponse();
+    Assertions.assertEquals(OzoneManagerProtocolProtos.Status.KEY_UNDER_LEASE_SOFT_LIMIT_PERIOD,
+        omResponse.getStatus());
+    Thread.sleep(2000);
+    // Call recovery after soft limit period it should succeed
+    omClientResponse = validateAndUpdateCache();
+    omResponse = omClientResponse.getOMResponse();
+    Assertions.assertEquals(OzoneManagerProtocolProtos.Status.OK, omResponse.getStatus());
+    RecoverLeaseResponse recoverLeaseResponse = omResponse.getRecoverLeaseResponse();
+    KeyInfo keyInfo = recoverLeaseResponse.getKeyInfo();
+    Assertions.assertNotNull(keyInfo);
+
+    // Call recovery again it should succeed
+    omClientResponse = validateAndUpdateCache();
+    omResponse = omClientResponse.getOMResponse();
+    Assertions.assertEquals(OzoneManagerProtocolProtos.Status.OK, omResponse.getStatus());
+    recoverLeaseResponse = omResponse.getRecoverLeaseResponse();
+    keyInfo = recoverLeaseResponse.getKeyInfo();
+    Assertions.assertNotNull(keyInfo);
```

Comment:
@ashishkumar50, could you test the both force and non-force case in this testLeaseSoftLimitForHsyncRecoverFile ?  Try the Parametrized test.

Reply:
Done